packages/contracts/src/contracts/current/protocol/Exchange/MixinWrapperFunctions.sol

```diff
@@ -16,281 +16,259 @@
-    /// @param orderAddresses Array of order's maker, taker, makerToken, takerToken, and feeRecipient.
-    /// @param orderValues Array of order's makerTokenAmount, takerTokenAmount, makerFee, takerFee, expirationTimestampInSec, and salt.
+    /// @param orders Array of orders.
     /// @param takerTokenFillAmounts Array of desired amounts of takerToken to fill in orders.
     /// @param signatures Maker's signatures of the orders.
     function batchFillOrdersNoThrow(
-        address[5][] orderAddresses,
-        uint[6][] orderValues,
-        uint[] takerTokenFillAmounts,
+        Order[] orders,
+        uint256[] takerTokenFillAmounts,
         bytes[] signatures)
         public
     {
-        for (uint256 i = 0; i < orderAddresses.length; i++) {
+        for (uint256 i = 0; i < orders.length; i++) {
             fillOrderNoThrow(
-                orderAddresses[i],
-                orderValues[i],
+                orders[i],
                 takerTokenFillAmounts[i],
                 signatures[i]
             );
         }
     }
 
     /// @dev Synchronously executes multiple fill orders in a single transaction until total takerTokenFillAmount filled.
-    /// @param orderAddresses Array of address arrays containing individual order addresses.
-    /// @param orderValues Array of uint256 arrays containing individual order values.
-    /// @param takerTokenFillAmount Desired total amount of takerToken to fill in orders.
+    /// @param orders Array of orders.
+    /// @param takerTokenFillAmount Desired amount of takerToken to fill.
     /// @param signatures Maker's signatures of the orders.
     /// @return Total amount of takerTokenFillAmount filled in orders.
     function marketFillOrders(
-        address[5][] orderAddresses,
-        uint256[6][] orderValues,
+        Order[] orders,
         uint256 takerTokenFillAmount,
         bytes[] signatures)
         public
         returns (uint256 totalTakerTokenFilledAmount)
     {
-        for (uint256 i = 0; i < orderAddresses.length; i++) {
-            require(orderAddresses[i][3] == orderAddresses[0][3]); // takerToken must be the same for each order
-            totalTakerTokenFilledAmount = safeAdd(totalTakerTokenFilledAmount, fillOrder(
-                orderAddresses[i],
-                orderValues[i],
-                safeSub(takerTokenFillAmount, totalTakerTokenFilledAmount),
-                signatures[i]
-            ));
-            if (totalTakerTokenFilledAmount == takerTokenFillAmount) break;
+        for (uint256 i = 0; i < orders.length; i++) {
+            require(orders[i].takerTokenAddress == orders[0].takerTokenAddress);
+            totalTakerTokenFilledAmount = safeAdd(
+                totalTakerTokenFilledAmount,
+                fillOrder(
+                    orders[i],
+                    safeSub(takerTokenFillAmount, totalTakerTokenFilledAmount),
+                    signatures[i]
+                )
+            );
```

Comment:
If stack is not a problem here, can we break out the safeAdd and safeSub into separate lines with new variables?